test/integration/fake_upstream.cc

```diff
@@ -66,7 +66,7 @@ void FakeStream::decodeData(Buffer::Instance& data, bool end_stream) {
 void FakeStream::decodeTrailers(Http::RequestTrailerMapPtr&& trailers) {
   absl::MutexLock lock(&lock_);
   setEndStream(true);
-  trailers_ = std::move(trailers);
+  trailers_.reset(trailers.release());
```

Comment:
Why is this reset/release needed?
Would it make more sense to have a similar code-pattern as done in `decodeHeaders()`?

Reply:
Hmmm.... I had to change trailers_ type, because it used to be a unique_ptr, which worked fine with trailers() returning a reference, but now when trailers() return a copy of a pointer it does not work with unique_ptr anymore.

So I assumed that I cannot move a unique_ptr into a shared_ptr, since those are different types. I just double checked, and it seems that std::shared_ptr actually does have a moving assignment from a unique_ptr, so this change is not actually needed.

Changing it back.